lib/api/src/js/module.rs

```diff
@@ -165,16 +200,28 @@ impl Module {
     /// This is safe since the JS vm should be safe already.
     /// We maintain the `unsafe` to preserve the same API as Wasmer
     pub unsafe fn from_binary_unchecked(
-        _store: &impl AsStoreRef,
-        binary: &[u8],
+        store: &impl AsStoreRef,
+        binary: impl IntoBytes,
```

Comment:
Why we want to own the bytes @john-sharratt ?

Reply:
so the compiled code can be shared to the main thread, compiled there and fed to the other web workers. its a complex setup to get it working because of the JS limitations

Reply:
if there is an easier way it would be good to drop the Bytes but it is tricky, thats for sure

Reply:
In the case of JS, the bytes are already stored in `Vec<u8>`. So it shall be possible to share the compiled code?